### ISSUE TITLE
Does belongsTo support a target key that is not primary?

### ISSUE THREAD

**Issue by Suiname** [sequelize/sequelize#5643]
I'm running into a problem with sequelize and I'm not sure if it's because it's not working as I expect or if it's not possible.  I looked at the documentation and it seems like it should work but couldn't find any examples or further detail.    I have two tables, User and UserJoin, the latter having a uuid column which is housing a UUID data type.  If I try to set the UUID field as the target key in the belongsTo relationship, I always get a foreign key constaint error when I sync() to create the tables.  If I make the UUID field as the primary key, then everything works as expected.  I'd prefer not to have the UUID be the primary key, is this possible?  

Here's the failing code:
```
var User = sequelize.define("User", {
first_name: { type: Sequelize.STRING, allowNull: false},
last_name: { type: Sequelize.STRING, allowNull: false},
//more data stuff
});

var UserJoin = sequelize.define("UserJoin", {
  first_name: { type: Sequelize.STRING, allowNull: false},
  last_name: { type: Sequelize.STRING, allowNull: false},
  uuid: { type: Sequelize.UUID, defaultValue: Sequelize.UUIDV1}
});

User.belongsTo(UserJoin, {foreignKey: "uuid", targetKey:"uuid"});
```

And Here's the code that works:
```
var User = sequelize.define("User", {
first_name: { type: Sequelize.STRING, allowNull: false},
last_name: { type: Sequelize.STRING, allowNull: false},
//more data stuff
});

var UserJoin = sequelize.define("UserJoin", {
  first_name: { type: Sequelize.STRING, allowNull: false},
  last_name: { type: Sequelize.STRING, allowNull: false},
  uuid: { type: Sequelize.UUID, defaultValue: Sequelize.UUIDV1, primaryKey: true}
});

User.belongsTo(UserJoin, {foreignKey: "uuid", targetKey:"uuid"});
```

**Comment by Etzos** [comment]
This is actually a constraint set by the database itself, not Sequelize.

With a little thought it's fairly obvious to determine why this is as well. You can't reference a row unambiguously if you aren't referencing something that has a unique constraint of some kind (e.g. there could be two rows in the UserJoin table with the same uuid, so which row should be referenced?). So, you can either mark the `uuid` column as a primary key or just make it unique. Since you don't want the `uuid` column to be a primary key, I recommend making it unique. That will solve the error you're getting.

tl;dr: This is not a bug. The database and Sequelize are working as intended. Thankfully, there is a workaround!

**Comment by Suiname** [comment]
That makes sense.  I'll try setting this as a unique key but not primary then.  Thanks.
